CMakeLists.txt

```diff
@@ -288,7 +288,19 @@ elseif(ONEDPL_BACKEND MATCHES "^(omp)$")
     find_package(OpenMP)
     if (OpenMP_CXX_FOUND)
         message(STATUS "Compilation for the host due to OpenMP backend.")
-        target_link_libraries(oneDPL INTERFACE OpenMP::OpenMP_CXX)
+        # Due to minor correctness issues with -fiopenmp / -Qiopenmp, we are using -fopenmp / -Qopenmp until they are corrected.
+        # Once correctness issues are resolved, we will limit this workaround to affected versions of specific compilers.
+        if (OpenMP_CXX_FLAGS MATCHES "^-fiopenmp$")
```

Comment:
I think we may want to detect any instance of `-fiopenmp` or `/Qiopenmp` or `-Qiopenmp` and if found replace the whole list with `-fopenmp` or `/Qopenmp` accordingly.  This will keep us closest to the options pre #974. We wouldn't want to only replace one suggested flag, as a second one may depend on the presence of the first.

This would mean that I should change the `MATCHES` lines to detect any instance of these flags, rather than if they are the only flag.